### PR TITLE
Use core api

### LINKED ISSUE #471: Connect Vote to the Core API
**Additional Description**
Some new features require advanced querying, and for this to happen in a safe and complient way we should integrate with the Core API.

**Acceptance Criteria**
1. Orgs
2. Persons
3. Roles
4. The required scopes need to be given to the Vote client

### PR DESCRIPTION
**Pull request by JakubC-projects**
# Change summary
1. Uses the core api
2. Dockerizes local setup

## Change type

-   [ ] No review
-   [ ] Small PR
-   [ ] Big PR
-   [x] Refactor

**Closes #471**

### CODE REVIEW

File: client/src/domain/Org.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 export interface Org {
```
Comment (JakubC-projects) [inline client/src/domain/Org.ts:1]: for some reason I couldn't import the sdk in the frontend to use it's types.
That's why we need to still have separate types for frontend

File: server/dev.dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
+FROM arangodb:3.8.6 as arango
+FROM node:14.15.5-buster as build
+LABEL MAINTAINER "Philip Dalen <philip.dalen@bcc.no>"
```
Comment (StevenMalaihollo) [inline server/dev.dockerfile:3]: Hmmmm

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    core_api:
+        profiles: ["all", "dependencies"]
+        image: ghcr.io/bcc-code/bcc-core-api/bcc-core-emulator:5daed245fb24d16f9f76de9adb7d2de7eebcc28e
+        volumes:
+            - ./core-api-test-data:/test-data
+        environment:
+            TEST_DATA_PATH: "/test-data"
+
+    firebase:
+        profiles: ["all", "dependencies"]
```
Comment (StevenMalaihollo) [inline docker-compose.yml:56]: image

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
+version: "3.8"
+
+services:
+    server:
```
Comment (StevenMalaihollo) [inline docker-compose.yml:4]: The server times out for me when composing up.

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        ports:
+            - 4040:4040
+            - 9229:9229
+        working_dir: /opt/membership/server
+        command: sh -c '/usr/bin/wait-for.sh firebase:4010 -- npm run dev:docker'
+        depends_on: ["arango", "core_api", "firebase"]
+        volumes:
+            - "/etc/timezone:/etc/timezone:ro"
+            - "/etc/localtime:/etc/localtime:ro"
+            - ./server:/opt/membership/server
```
Comment (StevenMalaihollo) [inline docker-compose.yml:24]: Lazy copying is never good 🥇

Comment (JakubC-projects) [reply docker-compose.yml:24]: guess you're right, fixed

File: server/dev.dockerfile (hunk 2 above)
Comment (JakubC-projects) [reply server/dev.dockerfile:3]: removed

File: scripts/compose-up.sh [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+docker-compose \
+    --profile all \
+    up --build -d --wait
```
Comment (StevenMalaihollo) [inline scripts/compose-up.sh:5]: Remove `--wait`

Comment (JakubC-projects) [reply scripts/compose-up.sh]: done

File: docker-compose.yml (hunk 4 above)
Comment (JakubC-projects) [reply docker-compose.yml:4]: should be fixed now

File: docker-compose.yml (hunk 3 above)
Comment (JakubC-projects) [reply docker-compose.yml:58]: fixed

**Pull request merged**